stl/inc/atomic

```diff
@@ -1000,7 +1000,12 @@ struct _Atomic_storage<_Ty, 8> { // lock-free using 8-byte intrinsics
     void store(const _TVal _Value) noexcept { // store with sequential consistency
         const auto _Mem           = _STD _Atomic_address_as<long long>(_Storage);
         const long long _As_bytes = _STD _Atomic_reinterpret_as<long long>(_Value);
+#if defined(__clang__) && defined(_M_IX86) // TRANSITION, LLVM-126516
+        static_assert(_M_IX86_FP != 0, "8 byte atomic store is not supported on clang-cl with /arch:IA32");
+        __atomic_store_n(_Mem, _As_bytes, __ATOMIC_SEQ_CST);
+#else // ^^^ workaround / no workaround vvv
         _ATOMIC_STORE_64_SEQ_CST(_Mem, _As_bytes);
+#endif // ^^^ no workaround ^^^

```

Comment:
This fix place looks superfluous. for seq_cst, the store is cmpxchg8b, it works.
We only need the other places fixed.

Reply:
Oh, my bad. The `_ATOMIC_STORE_64_SEQ_CST` is implemented vial relaxed store followed by full fence, not the way I expected it to be implemented. 

This fix is absolutely needed too.